### PR TITLE
Fix svc system sig handling

### PR DESCRIPTION
**Pull request by daved**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-861" title="DX-861" target="_blank">BugDX-861</a>  Service does not shutdown by system signal properly
  </td></table>
  <br />

### CODE REVIEW

**Comment by daved** [comment]
Resolving this was significantly frustrated by the signal handling found in the captain package. Bug filed: https://activestatef.atlassian.net/browse/DX-886

File: cmd/state-svc/service.go [hunk 1]
```diff
@@ -69,13 +68,16 @@ func (s *service) Stop() error {
...
 	}
 
 	return nil
 }
 
-func (s *service) Wait() {
-	_ = s.ipcSrv.Wait()
+func (s *service) Wait() error {
+	if err := s.ipcSrv.Wait(); err != nil {
+		return errs.Wrap(err, "IPC server operating failure")
```
Comment (Naatan) [inline cmd/state-svc/service.go:80]:
```suggestion
		return errs.Wrap(err, "Failed to wait for ipc service")
```

Error was a bit generic.

File: internal/ipc/comm.go [hunk 2]
```diff
@@ -40,12 +39,10 @@ func getPing(ctx context.Context, c *Client) (string, error) {
...
 }
 
-func stopHandler(c io.Closer) RequestHandler {
+func stopHandler(stop func() error) RequestHandler {
 	return func(key string) (string, bool) {
 		if key == keyStop {
-			defer func() {
-				go c.Close()
-			}()
+			_ = stop()
```
Comment (Naatan) [inline internal/ipc/comm.go:45]: If the error is of no consequence why is it returning an error?

File: internal/ipc/server.go [hunk 3]
```diff
@@ -133,35 +144,37 @@ func (ipc *Server) Start() error {
...
-	return nil
 }
 
 func (ipc *Server) Wait() error {
-	ipc.wg.Wait()
-	return nil
+	var retErr error
+	for err := range ipc.errsc {
+		if err != nil && retErr == nil {
+			retErr = err
```
Comment (Naatan) [inline internal/ipc/server.go:165]: We could at least wrap the error messages. Might loose the error type, but you're already at risk of that anyway.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: cmd/state-svc/service.go (hunk 1 above)
Comment (daved) [reply cmd/state-svc/service.go:80]: The Start function returns errors only related to starting, Stop only related to stopping, and Wait returns errors related to normal operation. So it's not failing to wait, it's relaying a malfunction after the wait unblocks. I'm happy to include the suggestion if you still think the change is useful.

File: internal/ipc/server.go (hunk 3 above)
Comment (daved) [reply internal/ipc/server.go:165]: This is just a funneling of returned errors, so the wrapping was already performed.

**Review by Naatan — DISMISSED** [review]

**Comment by daved** [comment]
The uninstall failures are reported here: https://activestatef.atlassian.net/browse/DX-895

**Review by Naatan — APPROVED** [review]

**Pull request merged**